--- /usr/lib/python3.11/site-packages/compose/cli/docker_client.py.orig
+++ /usr/lib/python3.11/site-packages/compose/cli/docker_client.py
@@ -94,7 +94,7 @@ def docker_client(environment, version=None, tls_config=None, host=None,
     according to the same logic as the official Docker client.
     """
     try:
-        kwargs = kwargs_from_env(environment=environment, ssl_version=tls_version)
+        kwargs = kwargs_from_env(environment=environment)
     except TLSParameterError:
         raise UserError(
             "TLS configuration is invalid - make sure your DOCKER_TLS_VERIFY "
